.github/workflows/proposal-pr-check.yml

```
@@ -0,0 +1,43 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+   changed_files:
+    runs-on: ubuntu-latest 
+    name: Test added-OFEP proposals for status
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get added files
+        id: added-files
+        uses: tj-actions/changed-files@v37
+        with: 
+          files: |
+            **.md
+      
+      - if: ${{ steps.added-files.outputs.added_files_count == 0 }}
+        name: No check needed
+        run: |
+          echo "No OFEP proposals were added"
+          exit 0
+        
+      - name: List all added files
+        id: list-added-files
+        if: ${{ steps.added-files.outputs.added_files_count != 0 }}
+        run: |
+          for file in ${{ steps.added-files.outputs.added_files }}; do
+            echo "$file was added"
+            echo "status=$(grep -e "State" "$file" | tr -d "## State: " | tr -d '"')" >> "$GITHUB_OUTPUT"
+          done
+
+      - if: ${{ (steps.added-files.outputs.added_files_count != 0) && (steps.list-added-files.outputs.status != 'REJECTED') && (steps.list-added-files.outputs.status != 'WITHDRAWN') && (steps.list-added-files.outputs.status != 'APPROVED') }}
```

```suggestion
      - if: ${{ steps.added-files.outputs.added_files_count > 0 && !contains(fromJSON('["REJECTED", "WITHDRAWN", "APPROVED"]', steps.list-added-files.outputs.status) }}
```

I haven't tested this but it should be possible to use `contains` for this.

https://docs.github.com/en/actions/learn-github-actions/expressions#example-matching-an-array-of-strings